DataFormats/Detectors/TRD/include/DataFormatsTRD/RawData.h

```diff
@@ -398,48 +401,50 @@ struct trdTestPattern6 {
 struct DigitMCMData {
   //             10987654321098765432109876543210
   // uint32_t:   00000000000000000000000000000000
-  /*  union {
-    uint32_t word0;
-    struct {
-      uint32_t a : 2;
-      uint32_t b : 5;
-      uint32_t adc : 21; //adc bit patternpad plane
-    } __attribute__((__packed__));
-  };*/
   union {
     //             10987654321098765432109876543210
     // uint32_t:   00000000000000000000000000000000
     uint32_t word;
     struct {
-      uint32_t c : 2; // c is wrong I cant remember name, but not a concern at the moment.
+      uint32_t f : 2;
       uint32_t z : 10;
       uint32_t y : 10;
       uint32_t x : 10;
     } __attribute__((__packed__));
   };
 };
 
+struct LinkToHCIDMapping {
```

Comment:
I guess this is a placeholder for the CCDB object that we want to have? Can you add a comment that this struct will be replaced soon?

Reply:
Actually I was planning on using this struct as CCDB object. I agree it is not super elegant and googling for bidirectional maps you do get quite some examples on how one could do it alternatively. But in the end this is an object which is loaded once in the beginning by the raw reader and which afterwards never changes during the run. It is 7kBytes which are kept in memory all the time. This should really not impact the performance. I am not sure if we should spent much time thinking about an elegant solution as long as what we have works and is also readable.

Reply:
We would update this object only when we have physically swapped links or if we notice in the future that more cables were not correctly plugged. We might see this from chambers which might be recovered in the future for example. Therefore we have this macro where we can just add a line for swapped links. Sure I could also put `swapLinks()` to the struct, there is no reason for keeping it in the macro.
The user will in the end anyhow only work with a `const` object received from the CCDB. So only `int getHCID(int link) const` and `int getLink(int hcid) const` will be used by the user

Reply:
Absolutely valid points. The only mis-leading item is then that the `isOK` method suggests a more thorough check: I would expect a forward and reverse mapping check for all links. Can you add a comment that this is a minimal check?

Reply:
I added this more thorough check now to the mapping